c++/include/orc/Common.hh

```diff
@@ -27,6 +27,30 @@
 #include <string>
 
 namespace orc {
+
+  class FileVersion {
```

Comment:
You should have 2 instance values for HIVE_0_11 and HIVE_0_12. With the current implementation, you'll need to throw if they use anything but HIVE_0_11, since that corresponds to RLEv1. We'll also need to have Reader.getFormatVersion() return this type too, but that can be done later.